public/js/discussion.js

```diff
@@ -0,0 +1,261 @@
+*
+* @method voteClickComment
+*/
+var voteClickComment = function(icon, vote) {
+    const commentId = icon.attr('id').split('_')[1];
+
+    $.ajax({
+        type: 'POST',
+        url: '/voteOnComment',
+        data: {
+            commentId: commentId,
+            vote: vote
+        },
+        success: function(data) {
+            if (data.voteValue === 1) {
+                $('#like_' + commentId).removeClass('liked-hover');
+                $('#like_' + commentId).addClass('liked');
+                $('#dislike_' + commentId).removeClass('disliked');
+            } else if (data.voteValue === -1) {
+                $('#dislike_' + commentId).removeClass('disliked-hover');
+                $('#like_' + commentId).removeClass('liked');
+                $('#dislike_' + commentId).addClass('disliked');
+            } else {
+                $('#like_' + commentId).removeClass('liked');
+                $('#dislike_' + commentId).removeClass('disliked');
+            }
+            $('#numLikes_' + commentId).html(data.likesCount);
+            $('#numDislikes_' + commentId).html(data.dislikesCount);
+        },
+        error: function(data){
+            if (data['status'] === 401) {
+                window.location.href = '/';
+            } else {
+                failSnackbar('Something went wrong');
+            }
+        }
+    });
+}
+
+/**
+* Sets the vote when the user clicks the thumbs-up/thumb-down icon on a reply
+*
+* @method voteClickReply
+*/
+var voteClickReply = function(icon, vote) {
+    const replyId = icon.attr('id').split('_')[1];
+
+    $.ajax({
+        type: 'POST',
+        url: '/voteOnReply',
+        data: {
+            replyId: replyId,
+            vote: vote
+        },
+        success: function(data) {
+            if (data.voteValue === 1) {
+                $('#like_' + replyId).removeClass('liked-hover');
+                $('#like_' + replyId).addClass('liked');
+                $('#dislike_' + replyId).removeClass('disliked');
+            } else if (data.voteValue === -1) {
+                $('#dislike_' + replyId).removeClass('disliked-hover');
+                $('#like_' + replyId).removeClass('liked');
+                $('#dislike_' + replyId).addClass('disliked');
+            } else {
+                $('#like_' + replyId).removeClass('liked');
+                $('#dislike_' + replyId).removeClass('disliked');
+            }
+            $('#numLikes_' + replyId).html(data.likesCount);
+            $('#numDislikes_' + replyId).html(data.dislikesCount);
+        },
+        error: function(data){
+            if (data['status'] === 401) {
+                window.location.href = '/';
+            } else {
+                failSnackbar('Something went wrong');
+            }
+        }
+    });
+}
+
+/**
+* Changes the color of the voting icon when the user hovers over it
+*
+* @method voteHover
+*/
+var voteHover = function(icon, type) {
```

Comment:
theres no CSS way for doing this?

Reply:
If we would to do it with CSS, it would disregard the fact if it is clicked or not.

Reply:
okay, just wondering!